torch/fx/experimental/symbolic_shapes.py

```diff
@@ -4107,8 +4137,10 @@ def create_symintnode(
         self,
         sym: sympy.Expr,
         *,
-        hint: Optional[int],
+        hint: Optional[Union[int, torch.SymInt]],
         source: Optional[Source] = None,
+        metafy_fn: Optional[Callable] = None,
+        nested_int_desc: Optional[MetaNestedIntDesc] = None,
```

Comment:
nit - putting on my dynamic shapes hat, it feels a bit bad that we're adding two args to this very-generic-function to make NJT happy. What do you think of:

(1) having the `metafy_fn` callable close over any state it needs (in this case the `MetaNestedIntDesc`), instead of requiring `create_symintnode()` to plumb it directly

(2) renaming `metafy_fn` to `nested_int_metafy_fn` (to make it clear that this is a nested_int` thing)

Reply:
Sounds much cleaner, let me try this out

Reply:
Updated.